bootstrap/pkg/apis/apps/kfconfig/types.go

```diff
@@ -368,3 +599,71 @@ func IsConditionNotFound(e error) bool {
 	return ok && err.Code == int(kfapis.NOT_FOUND) &&
 		strings.HasPrefix(err.Message, conditionNotFoundErrPrefix)
 }
+
+type SecretNotFound struct {
+	Name string
+}
+
+func (e *SecretNotFound) Error() string {
+	return fmt.Sprintf("Missing secret %v", e.Name)
+}
+
+func NewSecretNotFound(n string) *SecretNotFound {
+	return &SecretNotFound{
+		Name: n,
+	}
+}
+
+func IsSecretNotFound(e error) bool {
+	if e == nil {
+		return false
+	}
+	_, ok := e.(*SecretNotFound)
+	return ok
+}
```

Comment:
are we still using this?  should we migrate to `KfError`?

Reply:
discussed offline. let's migrate them later